test/server/browser-provider-test.js

```diff
@@ -384,5 +396,70 @@ describe('Browser provider', function () {
             });
         });
     });
+
+    describe('Remote provider', () => {
+        it('Should log an error if a browser window was not found', async () => {
+            const bc   = new BrowserConnectionMock();
+
+            bc.isReady = () => true;
+
+            const remoteProvider = proxyquire('../../lib/browser/provider/built-in/remote', {
+                'testcafe-browser-tools': {
+                    findWindow () {
+                        throw new Error('SomeError');
+                    }
+                },
+                debug: debugMock
+            });
+
+            const provider = new ProviderCtor(new BrowserProviderPluginHost(remoteProvider));
+
+            await provider.openBrowser(bc.id);
+
+            expect(debugMock.data['testcafe:browser:provider:built-in:remote']).eql('Error: SomeError');
+
+            debugMock.data = {};
+        });
+    });
+
+    it('Should raise a warning if a browser window was not found', async () => {
+        const bc      = new BrowserConnectionMock();
+
+        bc.isReady    = () => true;
+
+        let warning; let errorMsg;
```

Comment:
We try to always initialize variables in the declaration. Let's use `null` here to initialize them.